java/kafka/consumer/src/main/java/KafkaConsumerExample.java

```diff
@@ -22,6 +21,22 @@
 public class KafkaConsumerExample {
     private static final Logger log = LogManager.getLogger(KafkaConsumerExample.class);
 
+    public enum TracingSystem {
+        JAEGER,
+        OPENTELEMETRY;
+
+        public static KafkaConsumerExample.TracingSystem forValue(String value) {
+            switch (value) {
+                case "jaeger":
+                    return KafkaConsumerExample.TracingSystem.JAEGER;
+                case "opentelemetry":
+                    return KafkaConsumerExample.TracingSystem.OPENTELEMETRY;
+                default:
+                    return null;
+            }
+        }
+    }
```

Comment:
TBH, the way you use it now, I don't think the Enum really adds any value. I think you should either use it right from the beginning in the config class. Or you should just have two constants defining the two types and use `equals` to compare it with the value form config. I think this applies both here and in the other examples.

If you decide to keep it and use it from the config on, it might make sense to add maybe some `NONE` tracing to indicate that it is disabled instead of passing null around.

Reply:
If you keep the enum you can also do something like in https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/ClientDnsLookup.java#L21-L39 and rely on `valueOf()`